### DISCUSSION TITLE
feature/for-range/

### DISCUSSION THREAD

**Discussion by giscus** [cyub/go-1.14.13#2] [Announcements]
# feature/for-range/

遍历 - for-range语法 # for-range语法可以用来遍历数组、指向数组的指针，切片、字符串、映射和通道。
遍历数组 # 当遍历一个数组a时候，循环范围会从0到len(a) -1：
func main() { var a [3]int for i, v := range a { fmt.Println(i, v) } for i, v := range &a { fmt.Println(i, v) } } 遍历切片 # 当遍历一个切片s时候，循环范围会从0到len(s) -1，若切片是nil，则迭代次数是0次：
func main() { a := make([]int, 3) for i, v := range a { fmt.Println(i, v) } a = nil for i, v := range a { fmt.Println(i, v) } } for-range切片时候可以边遍历边append吗？ # 当遍历切片时候，可以边遍历边append操作，这并不会造成死循环。因为遍历之前已经确定了循环范围，遍历操作相当如下伪代码：

https://go.cyub.vip/feature/for-range/

**Comment by LewyHua** [comment]
map那里的第一行，写成了遍历切片

**Reply by cyub** [reply]
谢谢指出，已修正。
